### PR TITLE
week cases added

### PR DESCRIPTION
**Pull request by stoiandan**
Added support for the week case.

[Review on Reviewable](https://reviewable.io/reviews/m-zuber/myhome/119)

### CODE REVIEW

**Comment by M-Zuber** [comment]
Looks good.
Would you like to finish this by making sure that both the RecurringExpense and RecurringIncome have all the needed code?
Just make the changes and commit it to the branch. They will be automatically added to this PR, so there is no need to close and open a new PR

File: MyHome.UI/RecurringExpenseInput.cs [hunk 1]
```diff
@@ -272,6 +281,36 @@ private int CalcMonthsInRange()
...
+                int nWeeksInRange = this.CalcWeeksInRange();
+
+                // Sets a local variable that will hold the date of the individual income being saved
+                // the initial value is the start date
+                DateTime dtCurrentSaveDate = dtpStartDate.Value.Date;
+
+                //Loops for the amount of weeks in range
+                for (int nWeekIndex = 0; nWeekIndex < nWeeksInRange; nWeekIndex++)
+                {
+                    CreateNewIncome(dtCurrentSaveDate);
```
Comment (M-Zuber) [inline MyHome.UI/RecurringExpenseInput.cs]: This should be CreateNewExpense

**Comment by stoiandan** [comment]
I also noticed that by mistake I added some blank space, sorry,  got rid of it, and I rearranged the order in witch the method comes, that week follows after day and before month. The windows for both classes do have the same size so it should look decent.

File: MyHome.UI/RecurringExpenseInput.cs (hunk 1 above)
Comment (M-Zuber) [inline MyHome.UI/RecurringExpenseInput.cs]: This is in the Expense form, so you don't want to be saving the data as an income.
You want to change it to call `SaveNewExpense` instead.

File: MyHome.UI/RecurringIncomeInput.cs [hunk 2]
```diff
@@ -232,6 +238,8 @@ private void MultiDaySave()
             }
         }
 
+
+
         private void CreateNewIncome(DateTime dtCurrentSaveDate)
         {
             Income newIncome = new Income(decimal.Parse(txtAmount.Text), dtCurrentSaveDate, Convert.ToInt32(cmbCategory.SelectedValue), Convert.ToInt32(cmbPayment.SelectedValue), txtDetail.Text);
```
Comment (M-Zuber) [inline MyHome.UI/RecurringIncomeInput.cs]: This method does not actually save the income.
Add a call like the following:
`_incomeService.Create(newIncome)`

File: MyHome.UI/RecurringExpenseInput.cs (hunk 1 above)
Comment (stoiandan) [inline MyHome.UI/RecurringExpenseInput.cs]: So `CreateNewIncome` should be changed to CreateNewExpense? for week,year,month,day?

**Comment by M-Zuber** [comment]
It seems you forgot to add the week option to the switch in the RecurringExpenseInput form.
You can just copy it from the `RecurringIncomeInput` form

oh, and don't feel bad that you missed it. Thanks to this PR we caught a major bug that data wasn't being saved.

Otherwise the changes look great!
Thank you for all the hard work.

**Comment by stoiandan** [comment]
Does it look good now?

**Comment by M-Zuber** [comment]
:shipit: :+1: 
Looks great!!
Thank you very much!!

**Pull request merged**